R/html-tweak.R

```diff
@@ -99,22 +97,19 @@ tweak_tables <- function(html) {
   table <- xml2::xml_find_all(html, ".//table")
 
   if (length(table) != 0) {
-
-    existing <- xml2::xml_attrs(table, "class")
-    tweaked <- purrr::map(existing, prepend_class)
-
-    xml2::xml_attrs(table, "class") <- tweaked
+    tweaked <- purrr::walk(table, prepend_class, class = "table")
   }
 
   invisible()
 }
 
 prepend_class <- function(x, class = "table") {
-  if (!('class' %in% names(x))) {
-    c(class = class)
-  } else {
-    c(class = paste(class, x[["class"]]))
+
+  if (!is.na(xml2::xml_attr(x, "class"))) {
```

Comment:
Maybe this should be vectorised? I think this would work:

```
tweak_prepend_class <- function(x, class = "table") {
  cur <- xml2::xml_attr(x, "class")
  xml2::xml_attr(x, "class") <- ifelse(is.na(cur), cur, paste(cur, class))
  invisible()
}
```